test/powershell/Modules/Microsoft.PowerShell.Security/ConstrainedLanguageRestriction.Tests.ps1

```diff
@@ -0,0 +1,917 @@
+        }
+    }
+
+    Describe "Module scope scripts in constrained language mode" -Tags 'Feature','RequireAdminOnWindows' {
+
+        It "Verifies that while in constrained language mode script run in a module scope also runs constrained" {
+            Import-Module PSDiagnostics
+            $module = Get-Module PSDiagnostics
+
+            try
+            {
+                Invoke-LanguageModeTestingSupportCmdlet -EnableConstrainedLanguageMode
+
+                & $module { [object]::Equals("A", "B") }
+                throw "No Exception!"
+            }
+            catch
+            {
+                $exception = $_
+            }
+            finally
+            {
+                Invoke-LanguageModeTestingSupportCmdlet -EnableFullLanguageMode
+            }
+
+            $exception.FullyQualifiedErrorId | Should Match "MethodInvocationNotSupportedInConstrainedLanguage"
+        }
+    }
+
+    Describe "Switch -file in constrained language mode" -Tags 'Feature','RequireAdminOnWindows' {
+
+        It "Verifies that switch -file will not work in constrained language without provider" {
+
+            [initialsessionstate] $iss = [initialsessionstate]::Create()
+            $iss.LanguageMode = "ConstrainedLanguage"
+            [runspace] $rs = [runspacefactory]::CreateRunspace($iss)
+            $rs.Open()
+            $pl = $rs.CreatePipeline('switch -file c:\temp\foo.txt { "A" { "B" } }')
+
+            try
+            {
+                $pl.Invoke()
+                throw "No Exception!"
+            }
+            catch
+            {
+                $exception = $_
+            }
+            finally
+            {
+                $rs.Dispose()
+            }
+
+            $exception.FullyQualifiedErrorId | Should Match "DriveNotFoundException"
+        }
+    }
+
+    Describe "Get content syntax in constrained language mode" -Tags 'Feature','RequireAdminOnWindows' {
+
+        It "Verifies that the get content syntax returns null value in constrained language without provider" {
+
+            $iss = [initialsessionstate]::Create()
+            $iss.LanguageMode = "ConstrainedLanguage"
+            $rs = [runspacefactory]::CreateRunspace($iss)
+            $rs.Open()
+            $pl = $rs.CreatePipeline('${c:\temp\foo.txt}')
+
+            $result = $pl.Invoke()
+            $rs.Dispose()
+
+            $result[0] | Should Be $null
+        }
+    }
+
+    Describe "Stream redirection in constrained language mode" -Tags 'Feature','RequireAdminOnWindows' {
+
+        It "Verifies that stream redirection doesn't work in constrained language mode without provider" {
+
+            $iss = [initialsessionstate]::CreateDefault2()
+            $iss.Providers.Clear()
+            $iss.LanguageMode = "ConstrainedLanguage"
+            $rs = [runspacefactory]::CreateRunspace($iss)
+            $rs.Open()
+            $pl = $rs.CreatePipeline('"Hello" > c:\temp\foo.txt')

```

Comment:
Use testdrive

Reply:
It is not needed here as the test is expected to fail with specific error id.  Just a path string is required.

Reply:
same as above

Reply:
Done